client/src/PlayerLanding/PlayerInGame.tsx

```diff
@@ -24,61 +35,296 @@ export function PlayerInGame() {
   };
   const players = playerFilters[filter];
 
+  const [nightOrder, charactersByType] = React.useMemo(() => {
+    const allCharacters =
+      game.script
+        ?.map(({ id }) => ({ ...getRole(id), ...getRoleExtension(id) }))
+        .map((character) => character) ?? [];
+
+    const nightOrder = [...allCharacters]
+      .filter((character) => character.otherNight > 0)
+      .sort((a, b) => a.otherNight - b.otherNight);
+
+    return [
+      nightOrder,
+      {
+        Townsfolk: allCharacters.filter(({ team }) => team === "Townsfolk"),
+        Outsider: allCharacters.filter(({ team }) => team === "Outsider"),
+        Minion: allCharacters.filter(({ team }) => team === "Minion"),
+        Demon: allCharacters.filter(({ team }) => team === "Demon"),
+        // Unknown: allCharacters.filter(({ team }) => team === "Unknown"),
+      },
+    ];
+  }, [game.script]);
+
   return (
-    <Flex direction="column" gap="1" className="flex-1 overflow-hidden">
-      <Flex direction="column" gap="1" className="border-b border-red-700 p-2">
-        <Flex justify="between" p="2">
-          <Text className="capitalize">{me}</Text>
-          <Text className="capitalize">
-            {game.deadPlayers[me] ? "Dead" : "Alive"}
-          </Text>
-          <Text className="capitalize">
-            {!(game.deadPlayers[me] && game.deadVotes[me])
-              ? "Vote available"
-              : "Cannot vote"}
-          </Text>
-        </Flex>
+    <Tabs.Root
+      className="flex flex-1 flex-col"
+      value={selectedTab}
+      onValueChange={setSelectedTab}
+    >
+      <Tabs.List>
+        <Tabs.Trigger className="min-w-[100px] flex-1" value="script">
+          <motion.div layout>
+            <Text className="mr-1" color="red" asChild>
+              <GiScrollQuill />
+            </Text>
+          </motion.div>
+          <AnimatePresence>
+            <motion.div layout transition={{ duration: 0 }}>
+              {selectedTab === "script" && (
+                <motion.div
```

Comment:
Lying in bed I suspect now I got this and the above motion.div backwards. If swapping those works so I can remove the duration override this'll be way less ugly

Reply:
Actually if I'm right I don't think I need a animatepresence here either